src/components/displayAllComponent.js

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+import ChooseFolder from "./chooseFolder/chooseFolder";
+import { NavLink,Redirect } from "react-router-dom";
+import SvgSettingOptions from "./svgSetting/svgSettingOptions";
+import "../components/checkBoxSelection/checkBoxSelection.scss";
+import { changeObj, optimizeSvg } from "../../src/components/functions";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import {
+  ADD_VARIABLE,
+  UPDATE_VARIABLE,
+  DELETE_VARIABLE
+} from "../store/actionTypes";
+
+class DisplayAllComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+     redirect : false
+    };
+  }
+  getOptimizeSvg = async () => {
+    for (var option of this.props.svgOptions) {
+      changeObj(option, this.props.svgObject.plugins);
+    }
+    for (var setting of this.props.svgSettingList) {
+      if (this.props.svgOptions.length > 0) {
+        this.props.displayOptimize.push({
```

Comment:
create a constant or let variable for this object then push it into the array to prevent unintended changes from the shallow copy that would have been created by push